### DISCUSSION TITLE
error "You need a ISO C conforming compiler to use the glibc headers"

### DISCUSSION THREAD

**Discussion by changchenzhu** [NCAR/hrldas#138] [Q&A]
When I followed naohmp's technical documentation for installation and reached the step of running the "make all" compilation command, an error occurred. 

> cpp -P -traditional -DMPP_LAND -D_GFORTRAN_  -D_HRLDAS_OFFLINE_  -I/usr/include module_hrldas_netcdf_io.F > module_hrldas_netcdf_io.f90
> In file included from /usr/include/features.h:376:0,
>                  from /usr/include/errno.h:29,
>                  from /usr/include/netcdf.h:17,
>                  from module_hrldas_netcdf_io.F:2545:
> /usr/include/sys/cdefs.h:31:3: error: #error "You need a ISO C conforming compiler to use the glibc headers"
>  error "You need a ISO C conforming compiler to use the glibc headers"
>    ^
> make[1]: *** [module_hrldas_netcdf_io.o] Error 1
> make[1]: Leaving directory `/home/zcc/noahmp/hrldas/hrldas/IO_code'
> make: *** [all] Error 2
Initially, I thought it might be related to the version of the GCC compiler. Therefore, I attempted to use different versions of the GCC compiler, including 4.8.5, 7, and 10, but the issue persisted. At the moment, I am uncertain whether the problem is due to the version of my glibc. Currently, my glibc version is 2.17.

**Comment by cenlinhe** [comment]
looks like something is wrong with your netcdf installation. Did you use the same compiler when you install and compile the netcdf package?

**Reply by changchenzhu** [reply]
OK, thank you very much for your reply, I will check the problem you mentioned as soon as possible
